### PR TITLE
[SPARK-10164] [MLLIB] Fixed GMM distributed decomposition bug

### PR DESCRIPTION
**Pull request by jkbradley**
GaussianMixture now distributes matrix decompositions for certain problem sizes. Distributed computation actually fails, but this was not tested in unit tests.

This PR adds a unit test which checks this.  It failed previously but works with this fix.

CC: @mengxr

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixture.scala [hunk 1]
```diff
@@ -171,7 +171,7 @@ class GaussianMixture private (
 
     // Heuristic to distribute the computation of the [[MultivariateGaussian]]s, approximately when
     // d > 25 except for when k is very small
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixture.scala]: Move the comments to the method definition.

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixture.scala [hunk 2]
```diff
@@ -171,7 +171,7 @@ class GaussianMixture private (
...
-    val distributeGaussians = ((k - 1.0) / k) * d > 25
+    val distributeGaussians = GaussianMixture.distributeGaussians(k, d)
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/GaussianMixture.scala]: `distributeGaussians` -> `shouldDistributeGaussians`

File: mllib/src/test/scala/org/apache/spark/mllib/clustering/GaussianMixtureSuite.scala [hunk 3]
```diff
@@ -168,5 +182,9 @@ class GaussianMixtureSuite extends SparkFunSuite with MLlibTestSparkContext {
       Vectors.dense( 4.5605), Vectors.dense( 5.2043), Vectors.dense( 6.2734)
     )
 
+    val data2: Array[Vector] = Range(0, 25).map { i =>
```
Comment (mengxr) [inline mllib/src/test/scala/org/apache/spark/mllib/clustering/GaussianMixtureSuite.scala]: minor: maybe less code with `Array.tabulate`.

**Comment by jkbradley** [comment]
That should be it.

**Comment by jkbradley** [comment]
CC: @feynmanliang FYI this is the fix needed to run spark-perf

**Comment by feynmanliang** [comment]
LGTM thanks for the fix!

**Comment by jkbradley** [comment]
OK I'll merge this with master and branch-1.5

**Pull request closed without merging**